Release Automation Process (#9555)
geosolutions-it/MapStore2: .github/ISSUE_TEMPLATE/release_steps.md

## New stable branch creation

**Only** if you are need to create a new stable major release (YYYY.XX.00), you need first to create a branch for it. Follow the following:

- [ ] Run the [`Cut Release Branch`](https://github.com/geosolutions-it/MapStore2/actions/workflows/cut_major_branch.yml) workflow on github.

>       With the following Parameters:
>         - Use workflow from branch `master`
>         - MapStore branch name to use: `YYYY.XX.xx`
>         - Version of *MapFish Print*, *GeoStore* and *HTTP-Proxy* accordingly to the [MapStore release calendar](https://github.com/geosolutions-it/MapStore2/wiki/MapStore-Release-Calendars)
>         - use the default value for the other parameters

Reviewer: ![image](https://github.com/geosolutions-it/MapStore2/assets/11991428/c0b51c8a-d4b2-4e9c-ade6-451ddd6fe014)
not well formatted now

```suggestion
      With the following Parameters:
      
        - [ ] Use workflow from branch `master`
        - [ ] MapStore branch name to use: `YYYY.XX.xx`
        - [ ] Version of *MapFish Print*, *GeoStore* and *HTTP-Proxy* accordingly to the [MapStore release calendar](https://github.com/geosolutions-it/MapStore2/wiki/MapStore-Release-Calendars)
        - [ ] use the default value for the other parameters
```

> - [ ] Wait for the end of the process. At the end

Reviewer: ```suggestion
- [ ] Wait for the process to complete. At the end:
```

      - [ ] `npm run start:app`, then check that an empty homepage loads correctly
  - [ ] Test [Binary](http://build.geosolutionsgroup.com/view/MapStore/job/MapStore/view/MapStore%20QA/job/MapStore2-QA-Build/) (take the mapstore2-<RELEASE_BRANCH>-qa-bin.zip, from latest build)

## Prepare Release

- [ ] Run [`Prepare Release`](https://github.com/geosolutions-it/MapStore2/actions/workflows/pre_release.yml) workflow on github actions with the following parameters:
  - Use workflow from `branch` **YYYY.XX.xx** (the release branch)
  - Version to release **YYYY.XX.mm** (the effective number of the release)
  - MapStore version for changelog generation **YYYY.XX.mm** (the effective number of the previous release)
  - version to fix for the java module, accordingly with release schedule (e.g. `1.7.0`)
  - use the default value for the other parameters

> - [ ] Wait for the end of the process.At the end of this process:

Reviewer: ```suggestion
- [ ] Wait for the process to complete. At the end:
```

## Build and publishing release

- [ ] Run [`Create Release`](https://github.com/geosolutions-it/MapStore2/actions/workflows/create_release.yml) workflow on github actions with the following parameters:
  - Use workflow from `branch` **YYYY.XX.xx** (the release branch)
  - Version to release **YYYY.XX.mm** (the effective number of the release)
- [ ] Launch [MapStore2-Stable-Releaser](http://build.geosolutionsgroup.com/view/MapStore/job/MapStore/view/MapStore%20Stable/job/MapStore2-Stable-Releaser/) Jenkins job with
   - **YYYY.XX.mm** for the version
   - **YYYY.XX.xx** for the branch to build
- [ ] Wait the end of the 2 process

When the processes are finished, the release is ready to be published on github in draft mode.

- [ ] Open the new release in draft from [here](https://github.com/geosolutions-it/MapStore2/releases)
- [ ] Update the link to Docker in the release notes with the link to the latest stable release (search the new tag on [docker hub](https://hub.docker.com/r/geosolutionsit/mapstore2/tags) )
- [ ] Update the description of the release details
- [ ] Publish the release

> - [ ] create on [ReadTheDocs](https://readthedocs.org/projects/mapstore/) project the version build for `vYYYY.XX.mm` (click on "Versions" and activate the version of the tag, created when release was published)

Author: ```suggestion
- [ ] create on [ReadTheDocs](https://readthedocs.org/projects/mapstore/) project the version build for `vYYYY.XX.mm` (click on "Versions" and activate the version of the tag, created when release was published)
- [ ] Update `Default version` to point the release version in the `Advanced Settings` menu of the [ReadTheDocs](https://readthedocs.org/dashboard/mapstore/advanced/) admin panel
```
